src/formats/roam-json.ts

```diff
@@ -266,10 +355,48 @@ export class RoamJSONImporter extends FormatImporter {
 
 		if (json.children) {
 			for (const child of json.children) {
-				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true));
+				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true, '', this.oldestTimestamp, this.newestTimestamp));
+			}
+		}
+
+		// once processing children is completed, add the YAML to the top
+		// check if any YAML options are set
+		if (this.fileDateYAML || this.titleYAML) {
+			checkYAMLoptions = true;
+		}
+
+		// add YAML frontmatter, if enabled
+		// only run on the initial set, skip if child 
+		if (checkYAMLoptions && !isChild) {
+
+			let timeCreated = this.oldestTimestamp;
+
+			frontMatterYAML.push('---');
+
+			// if "add title" option enabled, quotes added to prevent errors in frontmatter
+			if (this.titleYAML) {
+				frontMatterYAML.push(`title: "${setTitleProperty}"`);
+			}
+
+			// if "timestamps" option enabled
+			if (this.fileDateYAML) {
+				// if create is missing, use updated
+				// if updated is missing, use current Date()
+				let TSFormat = 'YYYY-MM-DD HH:mm:ss';
+			
+				let formatUpdateDate = this.newestTimestamp ? moment(this.newestTimestamp).format(TSFormat) : moment(new Date()).format(TSFormat);
+				let formatCreateDate = timeCreated ? moment(timeCreated).format(TSFormat) : formatUpdateDate;
+
+				frontMatterYAML.push('created: ' + formatCreateDate);
+				frontMatterYAML.push('updated: ' + formatUpdateDate);
 			}
+
+			frontMatterYAML.push('---');
 		}
 
+		// Add frontmatter YAML to the top of the markdown array
+		markdown.unshift(frontMatterYAML.join('\n'));
```

Comment:
Can this be moved into the if-block above? Just below `frontMatterYAML.push('---');`?

Reply:
Done, better to have it there so it's not doing anything when the options aren't set for the YAML header